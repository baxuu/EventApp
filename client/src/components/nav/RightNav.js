import React from 'react';

import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

import { appMainColor, eventCardFontColor } from '../../helpers/colors';

const View = styled.ul`
  list-style: none;
  display: flex;
  flex-flow: row nowrap;

  @media (max-width: 768px) {
    position: fixed;
    flex-direction: column;
    background-color: ${appMainColor};
    transform: ${({ expand }) =>
      expand ? 'translateX(0)' : 'translateX(100%)'};
    top: 0;
    right: 0;
    height: 100vh;
    width: 200px;
    padding-top: 4.5rem;
    transition: transform 0.3s ease-in-out;
  }
`;

const StyledNavLink = styled(NavLink)`
  line-height: 3.5rem;
  text-align: center;
  font-size: 1.2rem;
  color: ${eventCardFontColor};
  padding: 0 1.5rem;
  text-decoration: none;

  @media (max-width: 768px) {
    text-align: left;
  }
`;

const RightNav = ({ expand, setExpanded }) => {
  return (
    <View expand={expand}>
      <StyledNavLink to='/events' onClick={() => setExpanded(!expand)}>
        Events
      </StyledNavLink>
      <StyledNavLink to='/create-event' onClick={() => setExpanded(!expand)}>
        Create Event
      </StyledNavLink>
    </View>
  );
};

export default RightNav;
