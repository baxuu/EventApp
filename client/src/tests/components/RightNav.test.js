import React from 'react';

import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';

import RightNav from '../../components/Nav/RightNav';

Enzyme.configure({ adapter: new Adapter() });

describe('Navbar', () => {
  it('snapshot test', () => {
    const RightNavWrapper = shallow(<RightNav />);
    expect(RightNavWrapper).toMatchSnapshot();
  });
});
