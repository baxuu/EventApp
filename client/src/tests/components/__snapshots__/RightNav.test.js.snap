// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot test 1`] = `ShallowWrapper {}`;
